src/commands/registries/azure/tasks/buildImageInAzure.ts

```diff
@@ -5,12 +5,24 @@
 
 import * as vscode from 'vscode';
 import { IActionContext, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import { KnownRunStatus, Run } from '@azure/arm-containerregistry';
 import { scheduleRunRequest } from './scheduleRunRequest';
+import { sleep } from '../../../../utils/sleep';
 
-export async function buildImageInAzure(context: IActionContext, uri?: vscode.Uri | undefined): Promise<void> {
+const WAIT_MS = 500;
+
+export async function buildImageInAzure(context: IActionContext, uri?: vscode.Uri | undefined): Promise<Run> {
     if (!vscode.workspace.isTrusted) {
         throw new UserCancelledError('enforceTrust');
     }
 
-    await scheduleRunRequest(context, "DockerBuildRequest", uri);
+    const getRun = await scheduleRunRequest(context, "DockerBuildRequest", uri);
+    let run = await getRun();
+    while (run.status === KnownRunStatus.Running) {
```

Comment:
I don't think it's possible for this to go on forever so I didn't initially add a timeout.

I know the ACR build command can set a [timeout](https://learn.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries/buildtasks?pivots=deployment-language-arm-template#buildtaskproperties-1) so I'm assuming there's an upper bound enforced by them as well (though I couldn't find documentation verifying that).

Since builds can take a long time I'm unsure what a reasonable timeout would be. Ideally it would match a potential ACR build enforced one I'd think. I'll keep digging

Reply:
Yeah, I'm not sure what a reasonable timeout would be. Some builds shouldn't take more than thirty seconds and some shouldn't take more than two hours. 😄

Reply:
I think it's reasonable to rely on the ACR to "timeout" or fail eventually. We could poll forever but that would only be because ACR is continuously returning a status of Running which would mean something is broken on their end. It seems reasonable to trust that ACR will behave appropriately. 

In the very unlikely case that we do poll forever because the ACR returns `succeeded` forever it shouldn't impact users much. They can still execute other commands and interact with VS Code completely normally. There will just be the small overhead of polling continuously happening in the background until they close VS Code.

If we were to set a timeout it sounds like it would have to be an extremely long time (4 or 5 hours?). I'm not opposed to that but think it's likely not worth it since the point of the timeout would be to cover a very unlikely ACR bug where it returns a status of `Running` forever. Covering that potential case that has a small impact comes at the tradeoff of potentially setting a timeout that's too short (for some users with really long builds) which IMO is more important to avoid.

Reply:
That sounds perfectly reasonable to me.